src/extension.ts

```diff
@@ -2414,6 +2438,85 @@ async function setup(context: vscode.ExtensionContext, progress?: ProgressHandle
+                });
+                if (result !== undefined) {
+                    await ext.getProjectOutline().setSearchTerm(result);
+                }
+            }
+        ),
+        vscode.commands.registerCommand(
+            "cmake.outline.clearFilter",
+            async () => {
+                await ext.getProjectOutline().setSearchTerm("");
+            }
+        ),
+        vscode.commands.registerCommand(
+            "cmake.outline.toggleBookmark",
+            async (node?: BaseNode) => {
+                if (!node) {
+                    return;
+                }
+
+                let nodeName: string = '';
+                if (node instanceof TargetNode) {
+                    nodeName = node.name;
+                } else if (node instanceof SourceFileNode) {
+                    nodeName = node.name;
+                } else if (node instanceof DirectoryNode) {
+                    nodeName = node.pathPart;
+                } else {
+                    nodeName = node.id;
+                }
+
+                const wasAdded = await ext.getBookmarksProvider().toggleBookmark(node);
+                const action = wasAdded ? 'added to' : 'removed from';
+                void vscode.window.showInformationMessage(`"${nodeName}" ${action} bookmarks`);
+                ext.getProjectOutline().refresh(); // Refresh the outline to show the bookmark icon change
+            }
+        ),
+        vscode.commands.registerCommand(
+            "cmake.outline.removeBookmark",
+            async (bookmarkNode?: any) => {
+                if (bookmarkNode?.bookmark) {
+                    await ext.getBookmarksProvider().removeBookmark(bookmarkNode.bookmark.id);
+                    void vscode.window.showInformationMessage(`Bookmark for "${bookmarkNode.bookmark.name}" removed`);
+                    ext.getProjectOutline().refresh(); // Refresh the outline to remove the bookmark from the list
+                }
+            }
+        ),
+        vscode.commands.registerCommand(
+            "cmake.outline.removeBookmarkInline",
+            async (node?: BaseNode) => {
+                if (!node) {
+                    return;
+                }
+                await ext.getBookmarksProvider().removeBookmark(node.id);
+                const nodeName = node instanceof TargetNode ? node.name : node instanceof SourceFileNode ? node.name : node instanceof DirectoryNode ? node.pathPart : node.id;
+                void vscode.window.showInformationMessage(`Bookmark for "${nodeName}" removed`);
+                ext.getProjectOutline().refresh(); // Refresh the outline to remove the bookmark from the list
+            }
+        ),
+        vscode.commands.registerCommand(
+            "cmake.bookmarks.clearAll",
+            async () => {
+                const result = await vscode.window.showWarningMessage(
+                    'Are you sure you want to clear all bookmarks?',
+                    'Yes', 'No'
```

Comment:
What are your thoughts @hanniavalera on this confirmation message and whether it is necessary? @bradphelan Curious on your thoughts as well. 

We often get feedback that VS Code can overload users with notifications, so sometimes we try to mitigate adding notifications. However, I can understand the usage here.